lib/server.js

```diff
@@ -32,9 +52,20 @@ function handleValue (value) {
 
   // Ignore event handlers. `onclick=${(e) => doSomething(e)}`
   // will become.           `onclick=""`
-  if (typeof value === 'function') return '""'
+  if (typeof value === 'function') return ''
   if (value === null || value === undefined) return ''
   if (value.__encoded) return value
+
+  if (typeof value === 'object') {
+    return Object.keys(value).map(function (key) {
+      if (BOOL_PROPS.indexOf(key) !== -1) {
+        return value[key] ? key + '="' + key + '"' : ''
+      }
+      var handled = handleValue(value[key])
+      return key + '="' + handled + '"'
+    }).filter(Boolean).join(' ')
+  }
```

Comment:
Replacing this block with a `.reduce` call  into a string should be slightly faster.